opentracing/src/main/java/opentracing/Span.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2016 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package opentracing;
+
+/**
+ * Represents an in-flight span in the opentracing system.
+ *
+ * <p>Spans are created by the {@link Tracer#buildSpan} interface.
+ */
+public interface Span {
+
+  /**
+   * Sets the end timestamp and records the span.
```

Comment:
>
> Per the start() debate, we do need some way to override the finish
> timestamp in order to do after-the-fact translation of span-like logs, etc.
> So, void finish(long timestamp)?
>
Brave, which is based on duration not endTs, as this in the span-scoped
tracer:
/**
* Completes the span, which took {@code duration} microseconds.
*/
public void finishSpan(long duration)